Index: git/omapfb.c
===================================================================
--- git.orig/omapfb.c	2009-11-17 20:28:31.000000000 +0100
+++ git/omapfb.c	2009-11-17 20:29:34.000000000 +0100
@@ -115,8 +115,8 @@
     gst_structure_get_int (structure, "width", &width);
     gst_structure_get_int (structure, "height", &height);
 
-    self->overlay_info.xres = MIN (self->varinfo.xres, width) & ~15;
-    self->overlay_info.yres = MIN (self->varinfo.yres, height) & ~15;
+    self->overlay_info.xres = width;
+    self->overlay_info.yres = height;
     self->overlay_info.xres_virtual = self->overlay_info.xres;
     self->overlay_info.yres_virtual = self->overlay_info.yres;
 
@@ -136,8 +136,8 @@
     self->plane_info.enabled = 1;
     self->plane_info.pos_x = 0;
     self->plane_info.pos_y = 0;
-    self->plane_info.out_width = self->varinfo.xres;
-    self->plane_info.out_height = self->varinfo.yres;
+    self->plane_info.out_width = width;
+    self->plane_info.out_height = height;
 
     GST_INFO_OBJECT (self, "plane info: width=%u, height=%u",
                      self->varinfo.xres, self->varinfo.yres);
