Index: git/omapfb.c
===================================================================
--- git.orig/omapfb.c	2009-11-17 20:29:34.000000000 +0100
+++ git/omapfb.c	2009-11-17 20:34:25.000000000 +0100
@@ -107,6 +107,8 @@
     GstOmapFbSink *self;
     GstStructure *structure;
     gint width, height;
+    gint par_n, par_d;
+    gint out_width, out_height;
 
     self = GST_OMAPFB_SINK (bsink);
 
@@ -114,6 +116,7 @@
 
     gst_structure_get_int (structure, "width", &width);
     gst_structure_get_int (structure, "height", &height);
+    gst_structure_get_fraction (structure, "pixel-aspect-ratio", &par_n, &par_d);
 
     self->overlay_info.xres = width;
     self->overlay_info.yres = height;
@@ -133,11 +136,22 @@
         return FALSE;
     }
 
+    /* scale to width */
+    out_width = self->overlay_info.xres;
+    out_height = (height * par_d * self->overlay_info.xres + width * par_n / 2) /
+      (width * par_n);
+    if (out_height > self->overlay_info.yres) 
+    {
+      out_height = self->overlay_info.yres;
+      out_width = (width * par_n * self->overlay_info.yres + height * par_d / 2) /
+        (height * par_d);
+    }
+
     self->plane_info.enabled = 1;
-    self->plane_info.pos_x = 0;
-    self->plane_info.pos_y = 0;
-    self->plane_info.out_width = width;
-    self->plane_info.out_height = height;
+    self->plane_info.pos_x = (self->overlay_info.xres - out_width) / 2;;
+    self->plane_info.pos_y = (self->overlay_info.yres - out_height) / 2;
+    self->plane_info.out_width = out_width;
+    self->plane_info.out_height = out_height;
 
     GST_INFO_OBJECT (self, "plane info: width=%u, height=%u",
                      self->varinfo.xres, self->varinfo.yres);
