From 5e84df43aff38ba4269d19306482f5568e569d69 Mon Sep 17 00:00:00 2001
From: Felipe Contreras <felipe.contreras@nokia.com>
Date: Wed, 25 Mar 2009 17:56:12 +0200
Subject: [PATCH 3/7] h264enc: add custom omx_setup

In preparation for more hacks for TI components.

Signed-off-by: Felipe Contreras <felipe.contreras@nokia.com>

diff --git a/omx/gstomx_base_videoenc.c b/omx/gstomx_base_videoenc.c
index 476318a..3325717 100644
--- a/omx/gstomx_base_videoenc.c
+++ b/omx/gstomx_base_videoenc.c
@@ -320,6 +320,9 @@ omx_setup (GstOmxBaseFilter *omx_base)
         }
     }
 
+    if (self->omx_setup)
+        self->omx_setup (GST_OMX_BASE_FILTER (self));
+
     GST_INFO_OBJECT (omx_base, "end");
 }
 
diff --git a/omx/gstomx_base_videoenc.h b/omx/gstomx_base_videoenc.h
index 9ddcd25..cc400c4 100644
--- a/omx/gstomx_base_videoenc.h
+++ b/omx/gstomx_base_videoenc.h
@@ -42,6 +42,7 @@ struct GstOmxBaseVideoEnc
     guint bitrate;
     gint framerate_num;
     gint framerate_denom;
+    GstOmxBaseFilterCb omx_setup;
 };
 
 struct GstOmxBaseVideoEncClass
diff --git a/omx/gstomx_h264enc.c b/omx/gstomx_h264enc.c
index adf4c9a..1e2947a 100644
--- a/omx/gstomx_h264enc.c
+++ b/omx/gstomx_h264enc.c
@@ -69,6 +69,20 @@ type_class_init (gpointer g_class,
 }
 
 static void
+omx_setup (GstOmxBaseFilter *omx_base)
+{
+    GstOmxBaseVideoEnc *self;
+    GOmxCore *gomx;
+
+    self = GST_OMX_BASE_VIDEOENC (omx_base);
+    gomx = (GOmxCore *) omx_base->gomx;
+
+    GST_INFO_OBJECT (omx_base, "begin");
+
+    GST_INFO_OBJECT (omx_base, "end");
+}
+
+static void
 settings_changed_cb (GOmxCore *core)
 {
     GstOmxBaseVideoEnc *omx_base;
@@ -118,6 +132,8 @@ type_instance_init (GTypeInstance *instance,
     omx_base_filter = GST_OMX_BASE_FILTER (instance);
     omx_base = GST_OMX_BASE_VIDEOENC (instance);
 
+    omx_base->omx_setup = omx_setup;
+
     omx_base->compression_format = OMX_VIDEO_CodingAVC;
 
     omx_base_filter->gomx->settings_changed_cb = settings_changed_cb;
-- 
1.7.0.2

