Index: tiopenmax-0.3.5/application/src/openmax_il/avplay/src/AvPlayTest.c
===================================================================
--- tiopenmax-0.3.5.orig/application/src/openmax_il/avplay/src/AvPlayTest.c	2008-07-15 22:55:25.000000000 +0200
+++ tiopenmax-0.3.5/application/src/openmax_il/avplay/src/AvPlayTest.c	2009-12-10 18:46:02.000000000 +0100
@@ -724,7 +724,7 @@
     }
     else if (atoi(argv[1]) == 15) {
         /* With WB AMR */
-        eError = WmaWmvTest(argc,argv);
+        /* eError = WmaWmvTest(argc,argv); */
         //Test_With_Video_new (argc, argv);
     }
 
Index: tiopenmax-0.3.5/application/src/openmax_il/avplay/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/application/src/openmax_il/avplay/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/application/src/openmax_il/avplay/src/Makefile	2009-12-10 18:45:22.000000000 +0100
@@ -5,8 +5,7 @@
 OMX_DEBUG ?= 0      # master switch: turn debug on or off
 
 SRC=\
-	AvPlayTest.c \
-	AvPlayTest_WMV.c
+	AvPlayTest.c 
 
 HSRC=$(wildcard ../inc/*)
 
