Index: tiopenmax-0.3.5/system/src/openmax_il/clock_source/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/system/src/openmax_il/clock_source/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/system/src/openmax_il/clock_source/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -36,7 +36,7 @@
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
 $(COMPONENT_LIB): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) -L$(BRIDGELIBDIR) -lbridge
+	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) $(LDFLAGS) -L$(BRIDGELIBDIR) -lbridge
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/system/src/openmax_il/lcml/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/system/src/openmax_il/lcml/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/system/src/openmax_il/lcml/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -36,7 +36,7 @@
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
 $(COMPONENT_LIB): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) -L$(BRIDGELIBDIR) -lbridge -lpthread
+	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) $(LDFLAGS) -L$(BRIDGELIBDIR) -lbridge -lpthread
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/system/src/openmax_il/omx_core/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/system/src/openmax_il/omx_core/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/system/src/openmax_il/omx_core/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -31,7 +31,7 @@
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
 $(COMPONENT_LIB): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) -ldl
+	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) $(LDFLAGS) -ldl
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/system/src/openmax_il/resource_manager_proxy/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/system/src/openmax_il/resource_manager_proxy/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/system/src/openmax_il/resource_manager_proxy/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -26,7 +26,7 @@
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
 $(COMPONENT_LIB): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) -L$(BRIDGELIBDIR) -lbridge -lpthread $(LOMXPERFLIB)
+	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) $(LDFLAGS) -L$(BRIDGELIBDIR) -lbridge -lpthread $(LOMXPERFLIB)
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/video/src/openmax_il/post_processor/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/video/src/openmax_il/post_processor/src/Makefile	2009-12-10 21:36:57.000000000 +0100
+++ tiopenmax-0.3.5/video/src/openmax_il/post_processor/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -40,7 +40,7 @@
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
 $(COMPONENT_LIB): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) -L$(OMXLIBDIR) -lpthread -lOMX_ResourceManagerProxy
+	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) $(LDFLAGS) -L$(OMXLIBDIR) -lpthread -lOMX_ResourceManagerProxy
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/video/src/openmax_il/video_decode/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/video/src/openmax_il/video_decode/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/video/src/openmax_il/video_decode/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -48,7 +48,7 @@
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
 $(COMPONENT_LIB): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) -L$(OMXLIBDIR) -ldl -lpthread -lOMX_ResourceManagerProxy
+	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) $(LDFLAGS) -L$(OMXLIBDIR) -ldl -lpthread -lOMX_ResourceManagerProxy
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/application/src/openmax_il/avplay/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/application/src/openmax_il/avplay/src/Makefile	2009-12-10 21:36:58.000000000 +0100
+++ tiopenmax-0.3.5/application/src/openmax_il/avplay/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -36,7 +36,7 @@
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
 $(COMPONENT_TEST): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -ldl -lOMX_Core -lpthread
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) $(LDFLAGS) -L$(OMXLIBDIR) -ldl -lOMX_Core -lpthread
 	
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/audio/src/openmax_il/aac_dec/tests/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/audio/src/openmax_il/aac_dec/tests/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/audio/src/openmax_il/aac_dec/tests/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -45,7 +45,7 @@
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
 $(COMPONENT_TEST): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -ldl -lOMX_Core -lpthread
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) $(LDFLAGS) -ldl -lOMX_Core -lpthread
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/video/src/openmax_il/video_decode/test/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/video/src/openmax_il/video_decode/test/Makefile	2009-12-10 21:36:57.000000000 +0100
+++ tiopenmax-0.3.5/video/src/openmax_il/video_decode/test/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -49,7 +49,7 @@
 
 $(COMPONENT_TEST): $(OBJ)
 #	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -L$(BRIDGELIBDIR) -ldl -lOMX_Core -lpthread -lOMX_VideoDecoder -lbridge -lOMX_ResourceManagerProxy
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -lbridge -lOMX_Core
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) $(LDFLAGS) -L$(OMXLIBDIR) -lbridge -lOMX_Core
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/audio/src/openmax_il/aac_dec/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/audio/src/openmax_il/aac_dec/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/audio/src/openmax_il/aac_dec/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -40,7 +40,7 @@
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
 $(COMPONENT_LIB): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) -L$(OMXLIBDIR) -lOMX_ResourceManagerProxy 
+	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) $(LDFLAGS) -L$(OMXLIBDIR) -lOMX_ResourceManagerProxy 
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/system/src/openmax_il/resource_manager/activity_monitor_stub/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/system/src/openmax_il/resource_manager/activity_monitor_stub/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/system/src/openmax_il/resource_manager/activity_monitor_stub/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -36,7 +36,7 @@
 	cp -f $(COMPONENT_LIB) $(COMPONENT_TARGET)
 
 $(COMPONENT_LIB): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) -L$(BRIDGELIBDIR) -lbridge -lpthread
+	$(CROSS)gcc $(CFLAGS) -shared -o $(COMPONENT_LIB) $(OBJ) $(LDFLAGS) -L$(BRIDGELIBDIR) -lbridge -lpthread
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/system/src/openmax_il/audio_manager/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/system/src/openmax_il/audio_manager/src/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/system/src/openmax_il/audio_manager/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -26,7 +26,7 @@
 	cp -f $(COMPONENT_SYS) $(COMPONENT_TARGET)
 
 $(COMPONENT_SYS): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_SYS) $(OBJ) -L$(BRIDGELIBDIR) -lbridge
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_SYS) $(OBJ) $(LDFLAGS) -L$(BRIDGELIBDIR) -lbridge
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/system/src/openmax_il/resource_manager/src/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/system/src/openmax_il/resource_manager/src/Makefile	2009-12-10 21:36:57.000000000 +0100
+++ tiopenmax-0.3.5/system/src/openmax_il/resource_manager/src/Makefile	2009-12-10 21:36:58.000000000 +0100
@@ -26,7 +26,7 @@
 	cp -f $(COMPONENT_SYS) $(COMPONENT_TARGET)
 
 $(COMPONENT_SYS): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_SYS) $(OBJ) -L$(BRIDGELIBDIR) -lbridge $(LOMXPERFLIB) -L$(OMXLIBDIR) -lRAM
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_SYS) $(OBJ) $(LDFLAGS) -L$(BRIDGELIBDIR) -lbridge $(LOMXPERFLIB) -L$(OMXLIBDIR) -lRAM
 
 $(SRC): $(HSRC)
 
Index: tiopenmax-0.3.5/video/src/openmax_il/post_processor/test/Makefile
===================================================================
--- tiopenmax-0.3.5.orig/video/src/openmax_il/post_processor/test/Makefile	2008-07-15 22:15:47.000000000 +0200
+++ tiopenmax-0.3.5/video/src/openmax_il/post_processor/test/Makefile	2009-12-10 21:38:49.000000000 +0100
@@ -36,7 +36,7 @@
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
 $(COMPONENT_TEST): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -ldl -lOMX_Core -lpthread
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) $(LDFLAGS) -L$(OMXLIBDIR) -ldl -lOMX_Core -lpthread
 	
 $(SRC): $(HSRC)
 
